### DISCUSSION TITLE
Why prompt users to install the pre-release version?

### DISCUSSION THREAD

**Discussion by Knifa** [microsoft/vscode-cpptools#12260] [General]
Setting up my environment from scratch here again and I'm wondering what the intent is with this pop-up is, exactly?

![image](https://github.com/microsoft/vscode-cpptools/assets/622619/a351fb4b-54e8-45df-abab-cdd98698b6b3)

It was added in https://github.com/microsoft/vscode-cpptools/pull/11569 but there's no reason given in the PR. Users who know what they're doing and want the pre-release version can already switch to it easily through the UI, no?

![image](https://github.com/microsoft/vscode-cpptools/assets/622619/d04e4aeb-70f9-4152-af37-3728f9e25d09)

For me, a prompt like this signals something like --- the stable version is outdated and actually you want to use the pre-release version normally, or something along these lines. Why would everyday users want to use the pre-release version otherwise, enough to prompt them about it?

**Comment by bobbrow** [comment]
> Users who know what they're doing and want the pre-release version can already switch to it easily through the UI, no?

We actually promoted pre-releases long before VS Code added official support for them in the UI.  When the UI was added we removed our promotion in favor of giving VS Code a chance to encourage users to install our pre-releases.  Unfortunately, participation in our pre-releases declined significantly.  We rely on feedback from users in the pre-release to help us identify issues we don't hit during our in-house testing and our stable release quality suffered as a result.  After discussing this with the VS Code team, we both concluded that adding our prompt back was the best solution for us.

> Why would everyday users want to use the pre-release version otherwise, enough to prompt them about it?

Many users like to have access to features and bug fixes as early as possible but are unaware that there is a way to get them before they are more widely available.  When more people install the pre-release and provide feedback to us, this has the added benefit of making our stable releases more stable and less likely to need major patches, so everyone benefits.

**Comment by Knifa** [comment]
Fair enough, thanks for explaining!
